Plugins~/Src/MeshSyncClientBlender/python/unity_mesh_sync_common.py

```diff
@@ -102,15 +170,250 @@ def draw(self, context):
+        directory = context.scene.meshsync_unity_project_path
+        #Validate that the path is a project
+        if msb_validate_project_path(directory) == False:
+            #Try to fill a valid path from an already running project
+            bpy.ops.meshsync.path_from_server('INVOKE_DEFAULT')
+        else:
+            bpy.ops.meshsync.install_meshsync('INVOKE_DEFAULT')
+
+        return {'FINISHED'}
+
+class MESHSYNC_OT_TryGetPathFromServer(bpy.types.Operator):
+    bl_idname = "meshsync.path_from_server"
+    bl_label = "Try to get the unity path from a running server"
+
+    def execute(self, context):
+        if not msb_context.is_editor_server_available:
+            bpy.ops.meshsync.prompt_project_path('INVOKE_DEFAULT')
+            return {'FINISHED'}
+
+        #Get the project path
+        msb_context.sendEditorCommand(2)
+        context.scene.meshsync_unity_project_path = msb_context.editor_command_reply
+        directory = context.scene.meshsync_unity_project_path
+
+        if msb_validate_project_path(directory) == False:
+            # Prompt user to enter a valid path
+            bpy.ops.meshsync.prompt_project_path('INVOKE_DEFAULT')
+        else:
+            bpy.ops.meshsync.create_server('INVOKE_DEFAULT')
+            
+        return {'FINISHED'}
+
+class MESHSYNC_OT_PromptProjectPath(bpy.types.Operator, ImportHelper):
+    bl_idname = "meshsync.prompt_project_path"
+    bl_label = "Unity Project"
+
+    filepath : bpy.props.StringProperty(name = "filepath")
+
+    def execute(self, context):
+
+        context.scene.meshsync_unity_project_path = self.filepath
+
+        if msb_validate_project_path(self.filepath) == False:
+            message = self.filepath + " is not a Unity project"
+            MS_MessageBox(message = message)
+            print(message)
+            return {'FINISHED'}
+        else:
+            bpy.ops.meshsync.install_meshsync('INVOKE_DEFAULT')
+
+        return {'FINISHED'}
+
+class MESHSYNC_OT_InstallMeshSync(bpy.types.Operator):
+
+    bl_idname = "meshsync.install_meshsync"
+    bl_label = "Try to get the unity path from a running server"
+
+    def get_meshsync_path(self):
+        script_file = os.path.realpath(__file__)
+        directory = os.path.dirname(script_file)
+        system = platform.system()
+        if system == 'Windows':
+            return directory+"\\MeshSyncClientBlender\\resources\\com.unity.meshsync"
+        elif system == 'Darwin':
+            return directory+"/MeshSyncClientBlender/resources/com.unity.meshsync"
+        elif system == 'Linux':
+            return directory+"/MeshSyncClientBlender/resources/com.unity.meshsync"
+
+    def add_meshsync_to_unity_manifest(self, path, entry):
+        file = open(path, "r+");
+        data = json.load(file);
+
+        # check if MeshSync is installed on selected project
+        # if not installed, install it
+        found = False
+
+        for package in data['dependencies']:
```

Comment:
Checking `package-lock.json` is probably more robust if we want to know whether MeshSync has been added or not

Reply:
Sounds good, will change this.

Reply:
done